### PR TITLE
Updated secrets and support for HTTPS

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

**Comment by normanjaeckel** [comment]
@peb-adr @gsiv Please have a look at this PR. Hopefully I have set all env vars corretly. Please be care that I change the default HTTPS behavior and enabled local HTTPS with a (insecure) self-signed cert. Maybe you have some hints for our cert generation code.

**Comment by normanjaeckel** [comment]
Hint: If the setup does not work because of password issues, try to use "openslides" as postgres password. I will have a look at this error later.

**Comment by normanjaeckel** [comment]
I merge this. Please ping me if there are any problems.

File: pkg/config/default-config.yml [hunk 1]
```diff
@@ -33,8 +33,8 @@ defaultEnvironment:
   DATASTORE_DATABASE_PORT: 5432
   DATASTORE_DATABASE_NAME: openslides
   DATASTORE_DATABASE_USER: openslides
-  DATASTORE_DATABASE_PASSWORD: openslides
-  DATASTORE_DATABASE_PASSWORD_FILE: /run/secrets/datastore_postgres_password
+  DATASTORE_DATABASE_USER_FILE: /run/secrets/postgres_user
```
Comment (peb-adr) [inline pkg/config/default-config.yml:36]: the user will be provided plain text, see https://github.com/OpenSlides/openslides-datastore-service/pull/164

File: README.md [hunk 2]
```diff
@@ -22,14 +19,25 @@ running. Check if you have to run docker as local user or as root.
...
-OpenSlides only runs over HTTPS. See [HTTPS](#HTTPS)
+This will give you a default setup with local self-signed SSL certificates. You
+will get [a browser warning and have to care yourself about checking the
+fingerprint of the
+certificate](https://en.wikipedia.org/wiki/Self-signed_certificate).
+
+[Below](#SSL-encryption) you find for more information and how to use
+[caddys](https://github.com/OpenSlides/OpenSlides/blob/master/proxy) integrated
+certificate retrieval or how to disable the proxy expecting SSL. Disabling SSL
+is only possible if you use an extra proxy in front of OpenSlides. Keep in mind
```
Comment (peb-adr) [inline README.md:35]: Disabling SSL is only ~~possible~~ feasible ...

File: pkg/config/default-config.yml [hunk 3]
```diff
@@ -65,8 +64,8 @@ defaultEnvironment:
   MEDIA_DATABASE_PORT: 5432
   MEDIA_DATABASE_NAME: openslides
   MEDIA_DATABASE_USER: openslides
-  MEDIA_DATABASE_PASSWORD: openslides
-  MEDIA_DATABASE_PASSWORD_FILE: /run/secrets/media_postgres_password
+  MEDIA_DATABASE_USER_FILE: /run/secrets/postgres_user
```
Comment (peb-adr) [inline pkg/config/default-config.yml:67]: media-service will fallback to `CONFIG` if it cant find `CONFIG_FILE`, see [here](https://github.com/OpenSlides/openslides-media-service/blob/8435aca151c7529542ee5e9ccbee7aa2f2201f3b/src/config_handling.py#L71)
So for uniformity we should only keep `MEDIA_DATABASE_USER`.

File: pkg/config/default-docker-compose.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ x-default-environment: &default-environment
   DATASTORE_DATABASE_PORT: {{ .DATASTORE_DATABASE_PORT}}
   DATASTORE_DATABASE_NAME: {{ .DATASTORE_DATABASE_NAME }}
   DATASTORE_DATABASE_USER: {{ .DATASTORE_DATABASE_USER }}
-  DATASTORE_DATABASE_PASSWORD: {{ .DATASTORE_DATABASE_PASSWORD }}
+  DATASTORE_DATABASE_USER_FILE: {{ .DATASTORE_DATABASE_USER_FILE }}
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:17]: see above

File: pkg/config/default-docker-compose.yml [hunk 5]
```diff
@@ -46,7 +45,7 @@ x-default-environment: &default-environment
   MEDIA_DATABASE_PORT: {{ .MEDIA_DATABASE_PORT }}
   MEDIA_DATABASE_NAME: {{ .MEDIA_DATABASE_NAME }}
   MEDIA_DATABASE_USER: {{ .MEDIA_DATABASE_USER }}
-  MEDIA_DATABASE_PASSWORD: {{ .MEDIA_DATABASE_PASSWORD }}
+  MEDIA_DATABASE_USER_FILE: {{ .MEDIA_DATABASE_USER_FILE }}
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:48]: see above

File: pkg/config/default-docker-compose.yml [hunk 6]
```diff
@@ -143,7 +142,8 @@ services:
     secrets:
       - auth_token_key
       - auth_cookie_key
-      - datastore_postgres_password
+      - postgres_user
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:145]: see above

File: pkg/config/default-docker-compose.yml [hunk 7]
```diff
@@ -163,7 +163,8 @@ services:
       - datastore-reader
       - postgres
     secrets:
-      - datastore_postgres_password
+      - postgres_user
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:166]: see above

File: pkg/config/default-docker-compose.yml [hunk 8]
```diff
@@ -185,7 +186,8 @@ services:
       - postgres
       - redis
     secrets:
-      - datastore_postgres_password
+      - postgres_user
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:189]: see above

File: pkg/config/default-docker-compose.yml [hunk 9]
```diff
@@ -312,7 +314,8 @@ services:
       - datastore-writer
       - postgres
     secrets:
-      - media_postgres_password
+      - postgres_user
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:317]: see above

File: pkg/config/default-docker-compose.yml [hunk 10]
```diff
@@ -389,12 +392,10 @@ secrets:
...
   manage_auth_password:
     file: ./secrets/manage_auth_password
-  datastore_postgres_password:
-    file: ./secrets/datastore_postgres_password
-  media_postgres_password:
-    file: ./secrets/media_postgres_password
-  vote_postgres_password:
-    file: ./secrets/vote_postgres_password
+  postgres_user:
+    file: ./secrets/postgres_user
```
Comment (peb-adr) [inline pkg/config/default-docker-compose.yml:396]: see above

**Review by peb-adr — COMMENTED** [review]
Hey @normanjaeckel ,
I'm sorry I didn't propoerly review this earlier.
So I left some comments mostly about, that we can provide the postgres user in plaintext to all relevant services. That's btw also tested in our OS4 beta setup.

Also I noticed this before.
The new version of the `openslides` client seems no longer to be able to connect to the server.
I see you changed something in the connect code. Are there implications on how to use/call the program from command line now?
I only get this error:
```bash
$ openslides-managetool-newsecrets-16ad83c set-password -u 1 -p asdf -a localhost:8000 
--password-file=.../secrets/manage_auth_password                                                              
Error: connecting to gRPC server: creating gRPC client connection with grpc.DialContext(): context deadline exceeded
```
while the last version worked fine:
```bash
$ openslides-managetool-main-7cd5dd3 set-password -u 1 -p asdf -a localhost:8000 --pass
word-file=.../secrets/manage_auth_password
```

**Comment by normanjaeckel** [comment]
@peb-adr Please see #118. I can not reproduce your connection issues.

**Pull request merged**